crates/miden-tx/src/prover/prover_host.rs

```diff
@@ -105,34 +104,138 @@ where
+                prev_map_value,
+                new_map_value,
+            ),
+
+            // Access witnesses should be in the advice provider at proving time.
+            TransactionEvent::AccountVaultBeforeAssetAccess { .. } => Ok(Vec::new()),
+            TransactionEvent::AccountStorageBeforeMapItemAccess { .. } => Ok(Vec::new()),
+
+            TransactionEvent::AccountAfterIncrementNonce => {
+                self.base_host.on_account_after_increment_nonce()
+            },
+
+            TransactionEvent::AccountPushProcedureIndex { code_commitment, procedure_root } => {
+                self.base_host.on_account_push_procedure_index(code_commitment, procedure_root)
+            },
+
+            TransactionEvent::NoteAfterCreated {
+                note_idx,
+                metadata,
+                recipient_digest,
+                note_script,
+                recipient_data,
+            } => {
+                let recipient_data = self.base_host.on_note_after_created(
+                    note_idx,
+                    metadata,
+                    recipient_digest,
+                    note_script,
+                    recipient_data,
+                )?;
+
+                // A return value of Some means recipient data was present, but the script was not
+                // and this should not happen at proving time.
+                if recipient_data.is_some() {
+                    Err(TransactionKernelError::other(
+                        "note script should be in the advice provider at proving time",
+                    ))
+                } else {
+                    // A return value of None means the note creation was handled.
+                    Ok(Vec::new())
                 }
             },
-            TransactionEventHandling::Handled(mutations) => Ok(mutations),
-        }
+
+            TransactionEvent::NoteBeforeAddAsset { note_idx, asset } => {
+                self.base_host.on_note_before_add_asset(note_idx, asset).map(|_| Vec::new())
+            },
+
+            TransactionEvent::AuthRequest { signature, .. } => {
+                if let Some(signature) = signature {
+                    Ok(self.base_host.on_auth_requested(signature))
+                } else {
+                    Err(TransactionKernelError::other(
+                        "signatures should be in the advice provider at proving time",
+                    ))
+                }
+            },
+
+            // This always returns an error to abort the transaction.
+            TransactionEvent::Unauthorized {
+                message,
+                salt,
+                output_notes_commitment,
+                input_notes_commitment,
+                account_delta_commitment,
+            } => Err(self.base_host.on_unauthorized(
+                message,
+                salt,
+                output_notes_commitment,
+                input_notes_commitment,
+                account_delta_commitment,
+            )),
```

Comment:
I think `Unauthorized` should never be triggered at proving time. So, maybe instead of calling `base_host.on_unauthorized()` we could just return `TransactionKernelError::other` with an appropriate message.

This would also probably mean that `on_unauthorized()` could be removed from the base host.